packages/tofu/src/inspectSource.js

```diff
@@ -309,46 +341,77 @@ function inspectPatternElementForDeclarations(node, keyPath = []) {
   }
 }
 
+/**
+ *
+ * @param {import('@babel/types').ObjectPattern} node
+ * @param {string[]} keyPath
+ * @returns {Declaration[]}
+ */
 function inspectObjectPatternForDeclarations(node, keyPath) {
   // if it has computed props or a RestElement, we cant meaningfully pursue any deeper
   // return the node with the current path
   const expansionForbidden = node.properties.some(
-    (prop) => prop.computed || prop.type === 'RestElement'
+    (prop) =>
+      /** @type {import('@babel/types').ObjectProperty} */ (prop).computed ||
+      prop.type === 'RestElement'
   )
   if (expansionForbidden) {
     return [{ node, keyPath }]
   }
-  // expand each property into a path, recursively
+  /** @type {Declaration[]} */
   let results = []
-  node.properties.forEach((prop) => {
-    const propName = prop.key.name
-    const child = prop.value
-    results = results.concat(
-      inspectPatternElementForDeclarations(child, [...keyPath, propName])
-    )
+  const properties = /** @type {import('@babel/types').ObjectProperty[]} */ (
```

Comment:
note to self: go back and take another look at why the type assertion is needed.